alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -190,6 +197,83 @@ class TransportIndexMonitorAction @Inject constructor(
         }
     }
 
+    private fun validateActionsAcrossTriggers(totalMaxActions: Int) {
+        if (totalMaxActions == AlertingSettings.DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS) return
+
+        scope.launch {
+            val monitors = getMonitors(client)
```

Comment:
If this is a brand new monitor creation, we do not consider the new monitor's actions. Also if we increase/decrease the number of actions for an existing monitor, we dont consider that as well.

Reply:
We need to append the current monitor to the monitors list.